connect/transforms/src/test/java/org/apache/kafka/connect/transforms/CastTest.java

```diff
@@ -74,6 +79,7 @@ public void castWholeRecordKeyWithSchema() {
 
         assertEquals(Schema.Type.INT8, transformed.keySchema().type());
         assertEquals((byte) 42, transformed.key());
+        xform.close();
```

Comment:
Sure, I'll do that in the next patch, but I think in some of them we run into the same issue as the `MockProducerTest` issue you identified above. I hope I did not misunderstand your point.